accounts/service.go

```diff
@@ -313,36 +313,35 @@ func (s *InterceptorService) UpdateAccount(ctx context.Context,
 		return nil, ErrAccountServiceDisabled
 	}
 
-	account, err := s.store.Account(ctx, accountID)
-	if err != nil {
-		return nil, fmt.Errorf("error fetching account: %w", err)
-	}
-
 	// If the expiration date was set, parse it as a unix time stamp. A
 	// value of -1 signals "don't update the expiration date".
+	var expiry fn.Option[time.Time]
```

Comment:
probably nothing for this PR as we want minimal changes (just wanted to know your take on that), I think it would be better to convert to optional types one layer higher and use btcutil.Amount for the accountBalance?

Reply:
but re the optional types one layer above, yes I think let's leave that to a separate PR 👍

Reply:
erg, actually - im just gonna add a commit to this pr for at least making the type for the service layer's UpdateAccount btcutil.Amount 👍

Reply:
but let's leave the optional stuff to a diff PR

Reply:
added commit at the end